stl/inc/xstring

```diff
@@ -3985,13 +3985,17 @@ public:
     constexpr void resize_and_overwrite(_CRT_GUARDOVERFLOW const size_type _Newsize, _Operation _Op) {
         if (_Mypair._Myval2._Myres < _Newsize) {
             _Reallocate_grow_by(
-                _Newsize - size(),
-                [](_Elem* const _New_ptr, const _Elem* const _Old_ptr, const size_type _Oldsize,
-                    const size_type _Newsize) { _Traits::copy(_New_ptr, _Old_ptr, (_STD min) (_Oldsize, _Newsize)); },
-                _Newsize);
+                _Newsize - size(), [](_Elem* const _New_ptr, const _Elem* const _Old_ptr, const size_type _Oldsize) {
+                    _Traits::copy(_New_ptr, _Old_ptr, _Oldsize + 1);
+                });
         }
 

```

Comment:
I've inspected exception safety impact on x64. It is negligible for cases where it is needed, and none when lambda obviously doesn't throw.

Suggest implementing this:
https://github.com/AlexGuteniev/STL/commit/c51c9596ee70a1dfc2104afb0543a0842a72cabb

Reply:
I'm hesitant to make any further changes without a maintainer's opinion.

Reply:
But then exception guarantee you test for is partial, and does not work if the callback started filling in the string before throwing; in this case it will result in a string with a broken invariant